src/event-bridge/event-bus.ts

```diff
@@ -120,9 +124,37 @@ abstract class EventBusBase<E extends EventBusRuleInput>
   readonly eventBusName: string;
   readonly eventBusArn: string;
 
+  readonly native: NativeIntegration<EventBusBase<E>>;
+
   constructor(readonly bus: aws_events.IEventBus) {
     this.eventBusName = bus.eventBusName;
     this.eventBusArn = bus.eventBusArn;
+
+    // Closure event bus base
+    const eventBusName = this.eventBusName;
+    this.native = <NativeIntegration<EventBusBase<E>>>{
+      bootstrap: (context: Function<any, any>) => {
+        this.bus.grantPutEventsTo(context.resource);
+      },
+      preWarm: (prewarmContext: NativePreWarmContext) => {
+        prewarmContext.eventBridge();
```

Comment:
hmm, this looks interesting. Does NativePreWarmContext have all AWS SDK?

Reply:
Any reason why we don't have `preWarm` instantiate any clients it needs and return them? `prewarmContext` can provide a cache which can be used to avoid duplicate clients?

Reply:
thats what it does, I've just added helpers for common clients. you can also add whatever.

```ts
preWarmConext.getOrInit({  key: "anything", init: () => "hi" });
preWarmConext.get("anything") // "hi"

preWarmConext.getOrInit(PrewarmClients.LAMBDA);
preWarmConext.get(PrewarmClients.LAMBDA.key) // lambda client
```

Reply:
the helpers correctly apply any props and ensure that all consumers use the same implementation.

```ts
  LAMBDA: {
    key: "LAMBDA",
    init: (key, props) => new Lambda(props?.clientConfigRetriever?.(key)),
  },
```